### ISSUE TITLE
E1136 unsubscriptable-object for Enum item with type hint for dictionary assignment

### ISSUE THREAD

**Issue by cdreimer-thewriter** [PyCQA/pylint#6832]
### Bug description

I have an Enum item with a type hint for the assigned dictionary comprehension in enums.py.

```python
class Numbers(Enum):
    FILE_RANK: Dict[Any, str] = {(96 + file, rank): chr(96 + file) + str(rank)
                                 for file in range(1, 9) for rank in range(1, 9)}
```

This Enum is accessed in board.py as:

```
Numbers.FILE_RANK.value[key]
```

Pylint complains that this Enum is an unsubscriptable object.

If I change ``FILE_RANK: Dict[Any, str]`` to ``FILE_RANK``, the error message goes away. If I run mypy, it complains about the missing type hint.

### Command used

```shell
pylint board.py
```

### Pylint output

```shell
board.py:152:23: E1136: Value 'Numbers.FILE_RANK.value' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

Pylint should ignore the type hint and not report an error.

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10 / VS Code 1.67.2

### Additional dependencies

mypy==0.960
mypy-extensions==0.4.3

**Comment by mbyrnepr2** [comment]
Thank you @cdreimer-thewriter. I believe this is resolved now via https://github.com/PyCQA/pylint/issues/7265.
I can no longer reproduce using your example with the latest astroid + pylint versions.
